segmentation/include/pcl/segmentation/impl/extract_clusters.hpp

```diff
@@ -61,21 +62,21 @@ pcl::extractEuclideanClusters (const PointCloud<PointT> &cloud,
   std::vector<int> nn_indices;
   std::vector<float> nn_distances;
   // Process all points in the indices vector
-  for (int i = 0; i < static_cast<int> (cloud.points.size ()); ++i)
-  {
+  auto lambda = [&](int &i) {
```

Comment:
The use of a lambda here looks funky to me.
I would say that what you're interested is a container wrapper, that it exposes either a filtered index view, or the full point clouds. There's something called CloudIterator in pcl common which might help achieve what you want.

Reply:
Thanks, that's exactly what I needed.